templates/js/link-unfurling/src/linkUnfurlingApp.js

```diff
@@ -0,0 +1,58 @@
+const { TeamsActivityHandler, CardFactory } = require("botbuilder");
+const helloWorldCard = require("./adaptiveCards/helloWorldCard.json");
+
+class LinkUnfurlingApp extends TeamsActivityHandler {
+  // Link Unfurling.
+  handleTeamsAppBasedLinkQuery(context, query) {
+    const previewCard = CardFactory.thumbnailCard("Preview Card", query.url, [
+      "https://raw.githubusercontent.com/microsoft/botframework-sdk/master/icon.png",
+    ]);
+
+    const attachment = { ...CardFactory.adaptiveCard(helloWorldCard), preview: previewCard };
+
+    return {
+      composeExtension: {
+        type: "result",
+        attachmentLayout: "list",
+        attachments: [attachment],
+        suggestedActions: {
+          actions: [
+            {
+              title: "default",
+              type: "setCachePolicy",
+              value: '{"type":"no-cache"}',
+            },
+          ],
+        },
+      },
+    };
+  }
+
+  // Zero Install Link Unfurling
+  handleTeamsAnonymousAppBasedLinkQuery(context, query) {
```

Comment:
Is this method totally the same as `handleTeamsAppBasedLinkQuery`?

Reply:
Yes. In my opinion, encapsulating the common part will make the logic more complex, and more complex to customize code. So I just leave it here.